scripts/bindings.py

```diff
@@ -177,36 +181,45 @@ def parse_section(name, filename):
     :return: string
     """
 
-    global sections
     section = sections[name]
 
     return parse_content(filename, section[0], section[1])
 
 
 # You should regularly check the sections for the target 'ui.h' file.
 
+# A section is from and including the first line that matches the first
+# text up to and including the first line that matches the second text.
+# Two or more words are used to make it more robust.
 sections = {
-    'main': [28, 46],
-    'control': [48, 87],
-    'window': [89, 97],
-    'button': [99, 104],
-    'box': [106, 113],
-    'entry': [115, 122],
-    'checkbox': [124, 131],
-    'label': [133, 137],
-    'tab': [139, 147],
-    'group': [149, 156],
-    'spinbox': [163, 168],
-    'progressbar': [170, 175],
-    'slider': [177, 181],
-    'separator': [183, 185],
-    'combobox': [187, 193],
-    'editablecombobox': [195, 202],
-    'radiobuttons': [204, 207],
-    'datetimepicker': [209, 213],
-    'multilineentry': [215, 225],
-    'menuitem': [227, 233],
-    'menu': [235, 248],
+    'main': ("uiInitOptions uiInitOptions;", "void uiFreeText(char"),
+    'control': ("uiControl uiControl;",
+                "void uiUserBugCannotSetParentOnToplevel(const"),
+    'window': ("uiWindow uiWindow;", "uiWindow *uiNewWindow"),
+    'button': ("uiButton uiButton;", "uiButton *uiNewButton"),
+    'box': ("uiBox uiBox;", "uiBox *uiNewVerticalBox"),
+    'entry': ("uiEntry uiEntry;", "uiEntry *uiNewEntry"),
+    'checkbox': ("uiCheckbox uiCheckbox/", "uiCheckbox *uiNewCheckbox"),
+    'label': ("uiLabel uiLabel;", "uiLabel *uiNewLabel"),
+    'tab': ("uiTab uiTab;", "uiTab *uiNewTab"),
+    'group': ("uiGroup uiGroup;", "uiGroup *uiNewGroup"),
+    'spinbox': ("uiSpinbox uiSpinbox;", "uiSpinbox *uiNewSpinBox"),
+    'progressbar': ("uiProgressBar uiProgressBar;",
+                    "uiProgressBar *uiNewProgressBar"),
+    'slider': ("uiSlider uiSlider;", "uiSlider *uiNewSlider"),
+    'separator': ("uiSeparator uiSeparator;",
+                  "uiSeparator *uiNewHorizontalSeparator"),
+    'combobox': ("uiCombobox uiCombobox;", "uiCombobox *uiNewCombobox"),
+    'editablecombobox': ("uiEditableCombobox uiEditableCombobox;",
+                         "uiEditableCombobox *uiNewEditableCombobox"),
+    'radiobuttons': ("uiRadioButtons uiRadioButtons;",
+                     "uiRadioButtons *uiNewRadioButtons"),
+    'datetimepicker': ("uiDateTimePicker uiDateTimePicker;",
+                       "uiDateTimePicker *uiNewTimePicker"),
+    'multilineentry': ("uiMultilineEntry uiMultilineEntry;",
+                       "uiMultilineEntry *uiNewNonWrappingMultilineEntry"),
+    'menuitem': ("uiMenuItem uiMenuItem;", "void uiMenuItemSetChecked"),
+    'menu': ("uiMenu uiMenu;", "void uiMsgBoxError"),
```

Comment:
I kind of like this approach as it is a little bit more robust than keeping track of the line numbers as I have. But as I said, this utility is just to be ran one time per "control" so that we can generate the function signatures. Then it is expected to just include new functions manually as they are added to libui..